test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+import org.opensearch.core.action.ActionListener;
+
+/**
+ * A simple implementation of {@link ActionListener} that captures the response and failures used for testing purposes.
+ *
+ * @param <T> the result type
+ */
+public class TestCapturingListener<T> implements ActionListener<T> {
```

Comment:
it can be local to the your test class

Reply:
This was required for other test files as well, I added this in another PR https://github.com/opensearch-project/OpenSearch/pull/14394 as well, which is merged. I haven't rebased this that's why this is coming up in the diff here.

Reply:
ok please rebase

Reply:
done